src/library/helpers/modal.js

```diff
@@ -36,13 +36,15 @@ var showModal = function(modalId, specialMsg, fixedPosition, closeFunc) {
   showOverlay(_closeFunc,modalId,fixedPosition);
 
   if (jQuery(modalId + ' .portal-pages-close').length === 0) {
-    jQuery(modalId).append('<a class="portal-pages-close">x</a>');
```

Comment:
matching the pathname isn't a good approach here. The path could easily change, or perhaps we will have nested modals in the signup process. It is more work, but I'd suggest adding a new argument to the showModal method, and a new option to the renderSignupModal method. This way when the portal calls this method it can tell it that the modal should not be closable.